packages/app/src/vis-packs/core/hooks.ts

```diff
@@ -136,7 +134,11 @@ export function useMappedArray<T>(
   autoScale?: boolean
 ) {
   const baseArray = useBaseArray(value, dims);
-  const mappedArray = useApplyMapping(baseArray, mapping);
+
+  const mappedArray = useMemo(
+    () => applyMapping(baseArray, mapping),
+    [baseArray, mapping]
+  );
```

Comment:
Since it's only used here and is not quite properly typed, I ended up inlining it.

Reply:
As a side remark, it would be nice to track other instances of `createMemo` to see if they also bring improper types. Removing them would also be a step towards shying away from unmaintained `react-use`.

Reply:
`createMemo` is not bad in itself, it just makes it harder to add signature overrides where needed. Fortunately `useApplyMapping` was the only remaining case of a `createMemo` being used to memoise a utility with signature overrides.